lib_xua/src/core/endpoint0/xua_endpoint0.c

```diff
@@ -459,22 +459,43 @@ static unsigned char hidReportDescriptorPtr[] = {
 
 #if _XUA_ENABLE_BOS_DESC
 /// Update the device interface GUID in both MSOS simple and composite descriptors
-static void update_guid_in_msos_desc(const char *guid_str)
+static void update_guid_in_msos_desc(const char *guid_str_dfu, const char *guid_str_control)
 {
     // composite descriptor
-    unsigned char *msos_guid_ptr = desc_ms_os_20_composite.msos_desc_registry_property.PropertyData;
+    unsigned char *msos_guid_ptr;
+#if XUA_DFU_EN
+    msos_guid_ptr = desc_ms_os_20_composite.msos_desc_registry_property_dfu.PropertyData;
     for(int i=0; i<DEVICE_INTERFACE_GUID_MAX_STRLEN; i++)
     {
-        msos_guid_ptr[2*i] = guid_str[i];
+        msos_guid_ptr[2*i] = guid_str_dfu[i];
         msos_guid_ptr[2*i + 1] = 0x0;
```

Comment:
not sure if we have mbstowcs() for this or not

Reply:
I couldn't get mbstowcs() to work. It seems to just do a copy. sizeof(wchar_t) is showing as 1 which also seems wrong.